TombEngine/Objects/TR3/Entity/tr3_tiger.cpp

```diff
@@ -99,100 +105,122 @@ namespace TEN::Entities::Creatures::TR3
-					item->Animation.TargetState = TIGER_STATE_IDLE;
-				else if (creature->Flags && AI.ahead)
-					item->Animation.TargetState = TIGER_STATE_IDLE;
-				else if (AI.bite && AI.distance < TIGER_RUN_ATTACK_RANGE)
-				{
-					if (LaraItem->Animation.Velocity.z == 0.0f)
+					}
+					else if (creature->Mood == MoodType::Escape &&
+						Lara.TargetEntity != item && AI.ahead)
+					{
 						item->Animation.TargetState = TIGER_STATE_IDLE;
-					else
-						item->Animation.TargetState = TIGER_STATE_RUN_SWIPE_ATTACK;
-				}
-				else if (creature->Mood != MoodType::Attack && TestProbability(TIGER_ROAR_CHANCE))
-				{
-					item->Animation.TargetState = TIGER_STATE_IDLE;
-					item->Animation.RequiredState = TIGER_STATE_ROAR;
-				}
-				else if (creature->Mood == MoodType::Escape &&
-					Lara.TargetEntity != item && AI.ahead)
-				{
-					item->Animation.TargetState = TIGER_STATE_IDLE;
-				}
-
-				creature->Flags = 0;
-				break;
-
-			case TIGER_STATE_BITE_ATTACK:
-			case TIGER_STATE_RUN_SWIPE_ATTACK:
-			case TIGER_STATE_POUNCE_ATTACK:
-				if (!creature->Flags && item->TouchBits.Test(TigerAttackJoints))
-				{
-					DoDamage(creature->Enemy, TIGER_ATTACK_DAMAGE);
-					CreatureEffect(item, TigerBite, DoBloodSplat);
-					creature->Flags = 1; // 1 = is attacking.
-				}
-
-				break;
+					}
+
+					creature->Flags = 0;
+					break;
+
+				case TIGER_STATE_BITE_ATTACK:
+				case TIGER_STATE_POUNCE_ATTACK:
+					if (!creature->Flags && item->TouchBits.Test(TigerBiteAttackJoints))
+					{
+						if	(item->Animation.AnimNumber == Objects[ID_TIGER].animIndex + TIGER_ANIM_BITE_ATTACK &&
+							 item->Animation.FrameNumber > g_Level.Anims[item->Animation.AnimNumber].frameBase + 4 ||
+							item->Animation.AnimNumber == Objects[ID_TIGER].animIndex + TIGER_ANIM_POUNCE_ATTACK_START &&
+							 item->Animation.FrameNumber > g_Level.Anims[item->Animation.AnimNumber].frameBase + 12 )
+						{
+							DoDamage(creature->Enemy, TIGER_BITE_ATTACK_DAMAGE);
+							CreatureEffect(item, TigerBite, DoBloodSplat);
+							creature->Flags = 1; // 1 = is attacking.
+						}
+					}
+					break;
+
+				case TIGER_STATE_RUN_SWIPE_ATTACK:
+					if (!creature->Flags && item->TouchBits.Test(TigerSwipeAttackJoints))
+					{
+						if (item->Animation.AnimNumber == Objects[ID_TIGER].animIndex + TIGER_ANIM_RUN_SWIPE_ATTACK &&
+							item->Animation.FrameNumber >= g_Level.Anims[item->Animation.AnimNumber].frameBase + 6 &&
+							item->Animation.FrameNumber < g_Level.Anims[item->Animation.AnimNumber].frameBase + 16)
+						{
+							DoDamage(creature->Enemy, TIGER_SWIPE_ATTACK_DAMAGE);
+							CreatureEffect(item, TigerBite, DoBloodSplat);
+							creature->Flags = 1; // 1 = is attacking.
+						}
+					}
+
+					break;
 			}
 		}
 
 		CreatureTilt(item, tiltAngle);
 		CreatureJoint(item, 0, extraHeadRot.y);
 		CreatureAnimation(itemNumber, headingAngle, tiltAngle);
+
+		auto radius = Vector2(object->radius, object->radius * 1.33f);
+		AlignEntityToSurface(item, radius);
```

Comment:
this alignes the enemy to surface, Right? For instance, it will align according to the slope?

Reply:
Yes it align the entity to any slope, the most example could be the crocodile from TR4 which use it.

Reply:
Yes, is the function that Sezz made to fix the Big_Scorpion alignment, I though we could keep reusing it for other entities.

Reply:
okie :)